docs/architecture-decisions/adr002-default-catalog-file-format.md

```diff
@@ -0,0 +1,245 @@
+        </p>
+        <p>
+          Backstage specific entities have an <code>apiVersion</code> that is
+          prefixed with <code>backstage.io/</code>.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>kind</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The high level entity type. In the first version of the software catalog,
+          we will focus on the <code>Component</code> kind.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>metadata</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          Metadata about the entity, such as its name, that are not directly part of
+          the entity specification itself. See below for more details about this structure.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>spec</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The actual specification data that describes the entity.
+        </p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+The `metadata` root field has the following nested structure.
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Required</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td valign="top"><code>name</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The name of the entity. Names must be unique per kind, within a given namespace.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>namespace</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The `name` of a namespace that the entity belongs to. This field is optional.
+        </p>
+        <p>
+          Namespaces may also be part of the catalog, and are <code>v1</code>/<code>Namespace</code>
+          objects, i.e. not Backstage specific but the same as in Kubernetes.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>description</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          A human readable description of the entity, to be shown in Backstage. May contain
+          markdown.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>labels</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          Labels are key/value pairs of that are attached to the entity, and their use is
+          identical to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/">kubernetes object labels</a>.
```

Comment:
Do we expect these to be consumed by the frontend and plugins? I'm thinking they would at most be a search-only feature if they should exist at all.

Reply:
Yes for consumption. Not exactly meant for search, but for classification, subselection, and filtering. Also extends to company-specific such dimensions. See the "motivation" section of the linked article. This would be an excellent place to put things like

- service tier
- hardcoded tc4x at Spotify
- lifecycle
- perhaps even owner, but that's very debatable

Reply:
I'm thinking this is how you would build up your system model on top of `Component` (and possibly `API` and `Resource`) objects, by tying them together in different ways. These would be indexed in the database so we can do queries such as (pseudo sql):
```sql
-- Everything in the catalog for a system
SELECT * FROM catalog WHERE metadata.labels.system = "my-system";

-- All service components
SELECT * FROM catalog WHERE kind = "Component" AND spec.type = "service";

-- All components owned by a team
SELECT * FROM catalog WHERE kind = "Component" AND metadata.labels.owner = "team@spotify.com";

-- All cloud resources for an owner
SELECT * FROM catalog WHERE kind = "Resource" AND metadata.labels.owner = "team@spotify.com";

-- All APIs exposed by a system
SELECT * FROM catalog WHERE kind = "Api" AND metadata.labels.system = "my-system";
```
Essentially, everything that we would select by could go into `metadata.labels`, while the annotations are plugin-specific. That's how I think about it, at least.

Reply:
Yeah I think going with the k8s "motivation" section makes sense, the only thing that worries me a bit is that we end up with a mix of labels that are internal to companies, and others that shared plugins would want to consume.

It would be nice if we could make this a place to ONLY put company specific stuff and the plugins can't make any assumption about what exists. As in plugins get to consume kind, name, namespace, annotations and the spec.